### PR TITLE
use _isnotzero(v)

### LINKED ISSUE #15: Problem with sparse arrays: missing values
I am not sure where to report it, but I think there is a problem with handling `missing` values for sparse arrays:
```
julia> ms = sparse([1, 2], [1, 2], [10, missing])
2×2 SparseMatrixCSC{Union{Missing, Int64}, Int64} with 2 stored entries:
 10  ⋅
  ⋅   missing

julia> md = Matrix(ms)
2×2 Matrix{Union{Missing, Int64}}:
 10  0
  0   missing

julia> sparse(md)
ERROR: TypeError: non-boolean (Missing) used in boolean context

julia> ms[1, 2] = missing
ERROR: TypeError: non-boolean (Missing) used in boolean context

julia> ms[1, 1] = missing
missing

julia> ms
2×2 SparseMatrixCSC{Union{Missing, Int64}, Int64} with 2 stored entries:
  missing  ⋅
 ⋅          missing

julia> x = spzeros(3)
3-element SparseVector{Float64, Int64} with 0 stored entries

julia> y = similar(x, Union{Int, Missing})
3-element SparseVector{Union{Missing, Int64}, Int64} with 0 stored entries

julia> y[1] = missing
ERROR: TypeError: non-boolean (Missing) used in boolean context

julia> y[1] = 10
10

julia> y[1] = missing
missing
```

CC @nalimilan

### PR DESCRIPTION
**Pull request by SobhanMP**
closes #15

replaces a bunch of `!iszero` with `_isnotzero` in order to handle `Missing`.

### CODE REVIEW

**Comment by SobhanMP** [comment]
the only question is whether findprev/findnext should specialize based on f. i would say that it makes sense that it does because it calls f.

**Comment by ViralBShah** [comment]
Where is `_isnotzero` defined?

**Comment by SobhanMP** [comment]
`_isnotzero(x)` is defined in in SparseArrays.jl:39, it's the function @dkarrasch added to fix the handle missings in #177

**Comment by ViralBShah** [comment]
Ah I see it now - turns out I had not done a `git pull`.

**Comment by SobhanMP** [comment]
should i add tests to it?

**Comment by ViralBShah** [comment]
Yes, let's get a few tests and merge.

cc @bkamins

**Comment by ViralBShah** [comment]
Is there any noticeable performance impact since some of these can be deep in an inner loop?

**Comment by SobhanMP** [comment]
it has an inline overload for ::number so i should be zero

File: test/issues.jl [hunk 1]
```diff
@@ -9,6 +9,18 @@ using InteractiveUtils: @which
 include("forbidproperties.jl")
 include("simplesmatrix.jl")
 
+
+@testset "issue #15" begin
+    s = sparse([1, 2], [1, 2], [10, missing])
+    d = Matrix(s)
+    @test (sparse(d); true)
+    s2 = sparse(d)
```
Comment (bkamins) [inline test/issues.jl:17]: I would rather test type of `s2`.

File: test/issues.jl [hunk 2]
```diff
@@ -9,6 +9,18 @@ using InteractiveUtils: @which
...
+    @test s2[1, 1] == 10
+    @test s2[2, 1] == 0
+    @test s2[1, 2] == 0
+    @test s2[2, 2] === missing
```
Comment (bkamins) [inline test/issues.jl:21]: maybe also add `setindex!` tests that were failing in https://github.com/JuliaSparse/SparseArrays.jl/issues/15?

**Comment by SobhanMP** [comment]
the remaining `iszero`s seem to be used on the indexes (which are always integer) so should not cause any problem.

**Review by ViralBShah — APPROVED** [review]

**Pull request merged**